### PR TITLE
[all] change char* with std::string in BaseObjectDescription

### PR DESCRIPTION
**Pull request by adagolodjo**
Nothing important, just switch char* with std::string in the component BaseObjectDescription. @damienmarchal @guparan 

______________________________________________________

This PR: 
- [ ] builds with SUCCESS for all platforms on the CI.
- [ ] does not generate new warnings.
- [ ] does not generate new unit test failures.
- [ ] does not generate new scene test failures.
- [ ] does not break API compatibility.
- [ ] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

**Comment by damienmarchal** [comment]
How brave you are :)

**Comment by adagolodjo** [comment]
Yeap @hugtalbot

File: SofaKernel/framework/sofa/core/objectmodel/BaseObjectDescription.h [hunk 1]
```diff
@@ -66,7 +66,7 @@ class SOFA_CORE_API BaseObjectDescription
 
     typedef std::map<std::string,Attribute> AttributeMap;
 
-    BaseObjectDescription(const char* name=nullptr, const char* type=nullptr);
+    BaseObjectDescription(const std::string name=nullptr, const std::string type=nullptr);
```
Comment (tgaugry) [inline SofaKernel/framework/sofa/core/objectmodel/BaseObjectDescription.h]: replace with `const std::string name=""` or `const std::string& name=""`
Underlying code likely has to be updated, I assume that it rely on `if(x == nullptr)` constructs.

File: SofaKernel/framework/sofa/core/objectmodel/BaseObjectDescription.h [hunk 2]
```diff
@@ -97,13 +97,13 @@ class SOFA_CORE_API BaseObjectDescription
...
-    virtual BaseObjectDescription* find(const char* nodeName, bool absolute=false);
+    virtual BaseObjectDescription* find(const std::string& nodeName, bool absolute=false);
 
     /// Find an object given its name (relative to this object)
-    virtual Base* findObject(const char* nodeName);
+    virtual Base* findObject(const std::string& nodeName);
 
     /// Get an attribute given its name (return defaultVal if not present)
-    virtual const char* getAttribute(const std::string& attr, const char* defaultVal=nullptr);
+    virtual const std::string getAttribute(const std::string& attr, const std::string& defaultVal=nullptr);
```
Comment (tgaugry) [inline SofaKernel/framework/sofa/core/objectmodel/BaseObjectDescription.h]: Maybe this should return a reference instead ?

**Review by tgaugry — COMMENTED** [review]
Both comments apply to all commits

**Comment by damienmarchal** [comment]
Anyone has plan for this PR to be updated and merged ?

**Pull request closed without merging**